pycaret/internal/tabular.py

```diff
@@ -7476,7 +7476,15 @@ def summary(show: bool = True):
         explainer = shap.TreeExplainer(model)
         logger.info("Compiling shap values")
         shap_values = explainer.shap_values(test_X)
-        shap_plot = shap.summary_plot(shap_values, test_X, show=show, **kwargs)
+
+        if len(shap_values) == 2:
+            try:
+                shap_plot = shap.summary_plot(shap_values[1], test_X, show=show, **kwargs)
+            except:
+                shap_plot = shap.summary_plot(shap_values, test_X, show=show, **kwargs)
+        else:
+            shap_plot = shap.summary_plot(shap_values, test_X, show=show, **kwargs)
+        
```

Comment:
```suggestion
        try:
        	assert len(shap_values) == 2
            shap_plot = shap.summary_plot(shap_values[1], test_X, show=show, **kwargs)
        except Exception:
            shap_plot = shap.summary_plot(shap_values, test_X, show=show, **kwargs)
        
```

Reply:
Thanks - Will this still check for len(shap_values) == 2 before falling back on exception?

Reply:
It will have the exact same logic as the code before. If the assertion is not met it will do the `except` block.

Reply:
Great thanks